app/code/download_data.py

```diff
@@ -19,4 +19,4 @@
 for one_a_tag in soup.findAll('a'):
     link = one_a_tag['href']
     if "Vacinados" in link:
-        urllib.request.urlretrieve(link, 'raw_db/' + link.split("/")[-1])
+        urllib.request.urlretrieve(link, 'app/data/raw/' + link.split("/")[-1])
```

Comment:
Acho que ficaria melhor mover o data/ pra raíz do código e o analytics pra dentro do data.

Reply:
@rbelem Poderia explicar os motivos do porque você acha que essa mudança ficaria melhor? Assim fica mais facil de entender a sua linha de raciocínio para essa mudança

Reply:
Acho que dessa forma ficariam melhor separados of códigos fonte dos dados.

Reply:
A proposta atual está assim:
- `app/analysis` (diretorio com as analysis geradas e utilizadas no front)
- `app/code` (diretorio com os codigos)
- `app/data/raw` (diretorio com os dados "cru" [pdf])
- `app/data/cleaned` (diretorio com os dados "limpos" [json])

E a sua proposta seria:
- `app/code/`
- `app/code/data/raw`
- `app/code/data/cleaned`
- `app/code/data/analysis`

é isso?

Reply:
@gilvan-reis seria mais ou menos assim:
- app/
- data/raw
- data/cleaned
- data/analysis

Reply:
Sobre ter o `data/analysis` não tenho nada contra (inclusive foi a minha primeira opção). Deixei separado porque considero que ele tem um contexto de crescimento diferente dos dado (atualmente todas as analises estão sendo exportadas em csv devido a forma como é utilizado, mas existem varias outras possibilidades para exportação dessas analises). Nesse ponto, vou esperar o comentário de mais pessoas para ter uma decisão.

Sobre ter o código no diretório `app` sou desfavorável a essa modificação quando comparado com ter um diretório especifico para isso (`app/code`), visto que ele especifica o conteúdo do diretório e evita a necessidade de uma nova reestruturação do código caso a quantidade de códigos cresça

Reply:
@gilvan-reis sobre o ```app/``` acho que quando o código começar a evoluir mais, será criados módulos dentro do ```app/```

No futuro ficaria assim:
- app/
- app/main.py
- app/parser/__init__.py
- app/parser/parser.py

Reply:
Ainda acho que a opção de ter uma pasta especifica é melhor quando comparado com a solução proposta, porque teríamos responsabilidades diferentes num mesmo nível de pasta. 

Vale reforçar que essa solução proposta é a mesma apresentada no comentário acima (mas sem a pasta `code`):
> E a sua proposta seria:
> 
> * `app/code/`
> * `app/code/data/raw`
> * `app/code/data/cleaned`
> * `app/code/data/analysis`
> 
> é isso?

Como a evolução do código é a mesma nos dois cenários, vou esperar outros comentários para ver se alguém tem outro argumento sobre essa mudança.

Reply:
Na conversa de hoje, chegamos numa idéia de que seria bom guardar todos os dados históricos já processados, para fins de transparência, o que acham de já pensarmos numa estratégia de organização de diretórios que englobe isso?

Reply:
Poderiamos colocar os arquivos gerados dentro de um diretorio no `analysis` contendo o nome do arquivo que gerou ele, mas o maior problema seria como o front iria acessar esses dados, visto que a solução atual é hard coded.

Tambem podemos pensar na possibilidade de separar as pastas por tipo de vacina, para permitir a adição de outros tipos no futuro

Reply:
@gilvan-reis oq vc acha de começarmos a separar nossos diretórios por dia? Assim, a estrutura passaria a ser mais ou menos assim

```bash
app
├── {...}
└── data
    ├── {...}
    ├── 2020-02-01
    │   ├── {...}
    │   ├── analysis
    │   ├── cleaned
    │   └── raw
    ├── 2020-02-02
    │   ├── {...}
    │   ├── analysis
    │   ├── cleaned
    │   └── raw
    ├── 2020-02-03
    │   ├── {...}
    │   ├── analysis
    │   ├── cleaned
    │   └── raw
    └── 2020-02-04
        ├── {...}
        ├── analysis
        ├── cleaned
        └── raw
```

Assim, o front poderia ter uma lista consolidada dessas datas (diretórios), e conforme fosse escolhida uma nova data lá, ele buscava no diretório correspondente

Reply:
@pvfrota é uma solução possível, mas as pastas "raw" e "cleaned" teriam apenas um arquivo cada. Se a opção do front ter essa lista de datas for aceita, podemos fazer na estrutura atual da mesma forma (apenas vai ser necessário subir uma PR no back e no front para cada alteração).

Acho que seria bom você também comentar o que acha sobre os dois pontos em aberto:
- Ter uma pasta "code" para centralizar os codigos ou deixa-los na raiz da pasta
- Colocar a pasta "analysis" dentro do "data" ou deixa-la na raiz
- EXTRA: os nomes das pastas estão bons? Fiz algumas alterações porque achei que estavam mais significativos, mas é importante todos terem essa mesma percepção.

Estou esperando essas opiniões para poder atualizar a PR e fazer o merge dela

Reply:
https://docs.python-guide.org/writing/structure/